test/e2e/providers/exec/docker.go

```diff
@@ -42,10 +44,14 @@ func RunDockerExecCommand(container, cmd, workdir string, envs map[string]string
 	}
 	args = append(args, "-w", workdir, container)
 
-	if strings.Contains(cmd, "/bin/sh") {
-		// Just split in to "/bin/sh" "-c" and "actual_cmd"
+	if strings.Contains(cmd, "/bin/sh") || strings.Contains(cmd, "bash") {
+		// Split in to tokens using shell-style rules for quoting and commenting.
 		// This is useful for passing piped commands in to os/exec interface.
-		args = append(args, strings.SplitN(cmd, " ", 3)...)
```

Comment:
If existing caller of this function relies on this strings.SplitN(cmd, " ", 3) behavior, the cmd string would be like: `/bin/sh -c aaa bbb ccc ddd`, and the caller expects that it is split into ["/bin/sh", "-c", "aaa bbb ccc"]. If we introduce a real shell lexer here, the same cmd string will be split into ["/bin/sh", "-c", "aaa", "bbb", "ccc"]. Sometimes this split result will not work. To keep the original behavior after we use a shell lexer, we need to find out all callers who pass a "/bin/sh -c blah blah" command string to this function, and update the command string to "/bin/sh -c 'blah blah'".